cmd/influxd/restore/restore.go

```diff
@@ -10,29 +10,53 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
+	"log"
 	"os"
 	"path/filepath"
 	"strconv"
+	"strings"
 
+	"compress/gzip"
+	"encoding/json"
+	"github.com/gogo/protobuf/proto"
 	"github.com/influxdata/influxdb/cmd/influxd/backup_util"
+	tarstream "github.com/influxdata/influxdb/pkg/tar"
 	"github.com/influxdata/influxdb/services/meta"
 	"github.com/influxdata/influxdb/services/snapshotter"
+	"github.com/influxdata/influxdb/tcp"
 )
 
+//go:generate protoc --gogo_out=. data.proto
+
 // Command represents the program execution for "influxd restore".
 type Command struct {
-	Stdout io.Writer
+	// The logger passed to the ticker during execution.
+	StdoutLogger *log.Logger
+	StderrLogger *log.Logger
+
+	// Standard input/output, overridden for testing.
 	Stderr io.Writer
+	Stdout io.Writer
 
-	backupFilesPath string
-	metadir         string
-	datadir         string
-	database        string
-	retention       string
-	shard           string
+	host string
+	path string
+
+	backupFilesPath     string
+	metadir             string
+	datadir             string
+	destinationDatabase string
+	sourceDatabase      string
+	backupRetention     string
+	restoreRetention    string
+	shard               string
```

Comment:
Probably update this to be a `uint` or `int` while we're in here updating things.